tmt/steps/provision/__init__.py

```diff
@@ -1190,6 +1188,10 @@ def _ssh_options(self) -> Command:
                 options.extend(['-i', key])
         if self.password:
             options.extend(['-oPasswordAuthentication=yes'])
+        if self.debug_level == 0:
+            # Avoid noisy ssh client messages spoiling test output (#2429).
+            # Enabled only when debug mode is not set.
+            options.extend(['-oLogLevel=QUIET'])
```

Comment:
So... that means that the `log.txt` will be missing ssh warnings and/or errors unless at least one `--debug` is provided. Could this possibly hide an interesting information which could be useful when inspecting errors? This would apply to Testing Farm runs as well as there `tmt run -v` is used, right?

Reply:
Only for test execution, that is the last agreement, see the comment thread with Milos ^.

Reply:
@psss I believe with that, if there are really some connection issues, we would hit them sooner and would provide the ssh errors.

Do you have any concerns with this last approach?

Reply:
I'm not sure this is a good approach, to be honest. If only ssh allowed to write its logging into separate file (which it doesn't)...

Imagine that 'test' somehow manipulates ssh server  - we will no idea what went wrong as no useful message will be in log.txt until one reschedules the job with debug level set.

How about removing the very last message from the log if it is 'Connection to X closed'  instead?

BTW the `Warning: Permanently added ...` message will go away with `StrictHostKeyChecking=no` (we don't verify nor store host keys so using this option should be OK)

Reply:
ok, so post-processing, we can try, will see where it goes

Reply:
> How about removing the very last message from the log if it is 'Connection to X closed' instead?

+1 for this, sounds like a one-liner fix?